### PR TITLE
Update Cambodia holidays: add Peace Day in Cambodia

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

- Add ["Peace Day in Cambodia" (ទិវា​សន្តិភាព​នៅ​កម្ពុជា​)](https://admin.taftac-cambodia.org/public/pdf_file/regulation_1704245695_Z7IALJjE.pdf) for Cambodia from 2024 onwards, English translation is based from the unofficial translation in the link above and https://www.acledabank.com.kh/kh/eng/md_publicholiday . (Simply referring to "Peace Day" alone could be confusing as Cambodia already has another ex-holiday called "Paris Peace Agreement's Day")
- More test case refractors as previously recommended in #2081 .

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new `holidays` functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've successfully run `make check`, all checks and tests are green

[contributing-guidelines]: https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/holidays/tree/dev/docs/source

### CODE REVIEW

File: holidays/countries/cambodia.py [hunk 1]
```diff
@@ -278,6 +279,16 @@ def _populate_public_holidays(self):
         self._add_holiday(bon_om_touk, _timedelta(bon_om_touk_date, -1))
         self._add_holiday(bon_om_touk, _timedelta(bon_om_touk_date, +1))
 
+        # ទិវា​សន្តិភាព​នៅ​កម្ពុជា​
```
Comment (KJhellico) [inline holidays/countries/cambodia.py:282]:
```suggestion
        # ទិវា​សន្តិភាព​នៅ​កម្ពុជា​
```
I think zero-width spaces is excessive (here and below in this name).

Comment (PPsyrius) [reply holidays/countries/cambodia.py:282]: Nice catch 👍

**Comment by PPsyrius** [comment]
I'm thinking of adding some kind of hidden char remover to the pre-commit later to prevent that from being included in the first place again. 👀

**Review by KJhellico — APPROVED** [review]
LGTM.

**Review by arkid15r — APPROVED** [review]
Thanks for the cleanup 👍

**Pull request merged**